### PR TITLE
logging: log_backend_rtt: Fix panic behavior when no host

### LINKED ISSUE #12051: LOG_PANIC never returns if RTT is disconnected
**Describe the bug**
When RTT is connected LOG_PANIC ends at some point. When RTT is disconnected it does not.

**To Reproduce**
Use LOG_PANIC with JLink disconnected.

**Expected behavior**
LOG_PANIC should never lock.

**Impact**
LOG_PANIC deadlocks so application cannot finish its activities.

**Environment (please complete the following information):**
- rev `5ea8454e46208ab3fd7006e98b47ca15d5c7a450`

### PR DESCRIPTION
**Pull request by nordic-krch**
RTT log backend was locking in panic if host absense was not yet
detected (early panic). It is fixed by adding detection of host
absense while pending on data being read by the host.

Fixes #12051.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
fyi: @pkral78

**Review by Qbicz — APPROVED** [review]
Tested on PCA20041. Device no longer hangs when going to deep sleep mode. Fixes #12051

File: subsys/logging/log_backend_rtt.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ static int drop_cnt;
 static int drop_warn;
 static int panic_mode;
 
-static bool host_present;
+static volatile bool host_present;
```
Comment (pdunaj) [inline subsys/logging/log_backend_rtt.c]: why volatile is needed?

Comment (nordic-krch) [reply subsys/logging/log_backend_rtt.c]: removed.

**Review by pdunaj — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**